### PR TITLE
Add errors stabilized in 1.34 to the core Error shims

### LINKED ISSUE #324: Support TryFromIntError in no-std
It's behind the futures feature gate, but futures need the FromString which is behind the std gate.

### PR DESCRIPTION
**Pull request by shepmaster**
Fixes #324

### CODE REVIEW

**Comment by FedericoPonzi** [comment]
Thanks a lot for the quick pr! Unfortunately I cannot test it because it's not building, I'm getting the same error from the CI

File: src/no_std_error.rs [hunk 1]
```diff
@@ -25,21 +25,16 @@ macro_rules! impl_error {
...
-    core::num::TryFromIntError,     // 1.34
+    core::num::ParseFloatError,     // 1.0
+    core::num::ParseIntError,       // 1.0
+    core::str::ParseBoolError,      // 1.0
+    core::str::Utf8Error,           // 1.0
+    core::char::DecodeUtf16Error,   // 1.9
+    core::fmt::Error,               // 1.11
+    core::cell::BorrowError,        // 1.13
+    core::cell::BorrowMutError,     // 1.13
+    core::char::ParseCharError      // 1.20
```
Comment (FedericoPonzi) [inline src/no_std_error.rs:36]: Missing comma here!

File: src/no_std_error.rs [hunk 2]
```diff
@@ -25,21 +25,16 @@ macro_rules! impl_error {
...
     core::array::TryFromSliceError, // 1.34
     core::char::CharTryFromError    // 1.34
```
Comment (FedericoPonzi) [inline src/no_std_error.rs:38]: And missing comma here

**Comment by shepmaster** [comment]
> because it's not building

Well, that's embarrassing. Let's try that again.

**Pull request merged**